keptn/lifecycle-toolkit
# feat: create an helm overlay

closes #600

# This PR

- modifies MAKEFILES to allow generation of helm overlay
- adds helm folder and overlay files 
- adds commands to generate CRDs into a separate crds folder 
- refactor operators to use a namespace taken from an ENV var
- shows an example configuration to patch and env variable 
- patches the namespaces with the helm release.namespace   

# How to test

- build your own images 
- then run `make helm-package RELEASE_REGISTRY=<yours> TAG=<yours>`
- this will generate an up to date helm chart in helm/chart/charts
- ` cd helm/chart && helm upgrade --install toolkit charts/keptn-lifecycle-toolkit-v0.5.0.tgz --namespace keptn-lifecycle --create-namespace`
- run e2e tests or integration tests to verify the installation  `make integration-test`
